app/services/midata_service.rb

```diff
@@ -21,12 +21,12 @@ def fetch_camp_unit_data(id)
     JSON.parse(self.class.get("/events/#{id}.json", query: auth_params).body)
   end
 
-  def fetch_camp_unit_data_hierarchy(root_id)
+  def fetch_camp_unit_data_hierarchy(root_id, root = true)
     root_data = fetch_camp_unit_data(root_id)
     children_ids = root_data.dig('events', 0, 'links', 'sub_camps')
 
-    return root_data if children_ids.nil?
+    return root_data if children_ids.nil? && !root
 
-    [root_data, children_ids.map { |child_id| fetch_camp_unit_data_hierarchy(child_id) }].flatten
+    [root_data, children_ids.map { |child_id| fetch_camp_unit_data_hierarchy(child_id, false) }].flatten
```

Comment:
since this is recursive, it should only return the root-data, if it's not the very top one. otherwise - if there are no subcamps on a root-camp, it will return it anyways.